src/txmempool.cpp

```diff
@@ -447,6 +450,15 @@ void CTxMemPool::removeUnchecked(txiter it)
     BOOST_FOREACH(const CTxIn& txin, it->GetTx().vin)
         mapNextTx.erase(txin.prevout);
 
+    if (vTxHashes.size() > 1) {
+        vTxHashes[it->vTxHashesIdx] = vTxHashes[vTxHashes.size() - 1];
+        txiter newhashit = mapTx.find(vTxHashes[it->vTxHashesIdx].first);
```

Comment:
I avoided doing that since I didnt want to look up iterator invalidation in boost's multi index map, will do that now.